### PR TITLE
vertical widgets

### PR DESCRIPTION
**Pull request by krawczykkatarzyna**
Screen Shot 2023-11-23 at 00 17 31

Screen Shot 2023-11-20 at 13 50 49

### CODE REVIEW

**Review by melyndav — CHANGES_REQUESTED** [review]
Hi @krawczykkatarzyna nice start on these widgets. :)  

1. The text should not show as blue on hover. Only the outter border should have a hover state.
image

2. The agreement box shouldn't have a hover state since it's not interactable.
image

3. Font weight on the heading in the teams widget doesn't match Figma.
image

4. Padding doesn't match Figma.
image

5. The widgets need to be handled down to the smallest desktop size (1025px). The forced width is causing issues. 
image

6. When I create a new objective to add it in, the placeholder content from the design is showing and not the correct content. It's showing accuratly on the colony admin page. 
image

7. There is an additional space between the percentage and the % symbol.
image

**Comment by krawczykkatarzyna** [comment]
@melyndav please do not check this pull request, it is not ready yet

File: src/components/common/ColonyHome/ColonyHome.tsx [hunk 1]
```diff
@@ -79,6 +128,183 @@ const ColonyHome = () => {
...
+            value: (
+              <div className="flex w-full gap-6">
+                <div className="relative w-[9.375rem] -mt-1">
+                  <DonutChart
+                    data={chartData || []}
+                    hoveredSegment={hoveredSegment}
+                    setHoveredSegment={setHoveredSegment}
+                  />
+                </div>
+                <div className="flex flex-col items-start gap-[0.8rem]">
```
Comment (CzarekDryl) [inline src/components/common/ColonyHome/ColonyHome.tsx:181]: Is `items-start` needed?

File: src/components/v5/common/WidgetBox/WidgetBox.tsx [hunk 2]
```diff
@@ -47,19 +48,25 @@ const WidgetBox: FC<WidgetBoxProps> = ({
     </>
   );
 
+  const hoverStyles =
+    'transition-all sm:hover:border-gray-900 sm:hover:cursor-pointer';
```
Comment (CzarekDryl) [inline src/components/v5/common/WidgetBox/WidgetBox.tsx:52]: Is hover needed on `cursor-pointer`?

File: src/components/v5/common/WidgetBox/partials/EmptyWidgetState.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  onClick,
+}) => (
+  <div
+    className={clsx(
+      className,
+      'flex flex-col gap-1 items-center justify-center text-gray-600 text-sm',
+    )}
+  >
+    {formatText(title)}
+    <button
```
Comment (CzarekDryl) [inline src/components/v5/common/WidgetBox/partials/EmptyWidgetState.tsx:21]: I think we have TextButton component, please check if you can use it

File: src/components/v5/common/WidgetBox/partials/types.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { MessageDescriptor } from 'react-intl';
+
+export interface EmptyWidgetStateProps {
+  title: MessageDescriptor;
+  actionTitle: MessageDescriptor;
```
Comment (CzarekDryl) [inline src/components/v5/common/WidgetBox/partials/types.ts:5]: React.ReactNode

File: src/components/v5/shared/Link/types.ts [hunk 5]
```diff
@@ -1,9 +1,10 @@
 import { MessageDescriptor } from 'react-intl';
-import { LinkProps as LinkComponentProps } from 'react-router-dom';
+import { LinkProps as LinkComponentProps, To } from 'react-router-dom';
 
 import { SimpleMessageValues } from '~types';
 
-export interface LinkProps extends LinkComponentProps {
+export interface LinkProps extends Omit<LinkComponentProps, 'to'> {
+  to?: To;
```
Comment (CzarekDryl) [inline src/components/v5/shared/Link/types.ts:7]: Why is `to` optional? Link will be without href?

File: src/components/v5/shared/ProgressBar/ProgressBar.tsx [hunk 6]
```diff
@@ -43,7 +43,7 @@ const ProgressBar: FC<ProgressBarProps> = ({
       </div>
       <span className="text-3 text-gray-600 ml-3">{progress}</span>
       {additionalText && (
-        <span className="text-3 text-gray-600 ml-1 flex-shrink-0">
```
Comment (CzarekDryl) [inline src/components/v5/shared/ProgressBar/ProgressBar.tsx:46]: Did you check if it's working in all places?

**Review by arrenv — CHANGES_REQUESTED** [review]
Thank you @krawczykkatarzyna, great work. Some feedback:

1. The hover state seems to stick when switching from one team to another in the chart.

same-tooltip-showing


2. The tooltip appears off to the side initial in some cases.

hover-state-chart


3. There should be a spacing, i.e. stroke between each segment of the cart.

Design: 
image


4.  I am getting a re-rendering issue when hovering over each team.

rerendering-issue

5. The responsiveness in the chart seems to not be working.

image

6. There seems to be a padding issue at the bottom of the agreement widget at smaller desktop screen sizes. This was found by emulating the widget width with the activity feed in and a screen width at 1025px i.e. just before the breakpoint.

image

**Review by rdig — DISMISSED** [review]
- No reputation in any teams (the whole colony) should just show a gray circle, currently it does not show anything
Screenshot from 2023-11-23 13-28-00

- Long domain names break the ui. There needs to be a limit after which an ellipsis kicks in
Screenshot from 2023-11-23 13-28-55

**Comment by krawczykkatarzyna** [comment]
@arrenv @rdig all issues fixed

File: src/components/common/ColonyHome/ColonyHome.tsx [hunk 7]
```diff
@@ -87,6 +117,186 @@ const ColonyHome = () => {
...
+                    MetaColony from next Quarter?
+                  </span>
+                  <MessageNumber message={1} />
+                </div>
+                <p className="text-sm text-gray-600 line-clamp-3">
+                  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed
+                  do eiusmod tempor incididunt ut labore et dolore magna aliqua.
+                  Ut enim ad minim veniam, quis nostrud exercitation ullamco
+                  laboris nisi ut aliquip ex ea velt.
+                </p>
```
Comment (arrenv) [inline src/components/common/ColonyHome/ColonyHome.tsx:277]: We can remove the hardcoded default data here.

File: src/components/common/ColonyHome/ColonyHome.tsx [hunk 8]
```diff
@@ -87,6 +117,186 @@ const ColonyHome = () => {
...
+                  toggleActionSidebarOn({
+                    [ACTION_TYPE_FIELD_NAME]: ACTION.CREATE_DECISION,
+                  })
+                }
+              />
+            ),
+            contentClassName: 'w-full',
+            className:
+              'flex-col p-6 bg-base-white h-[11.25rem] sm:cursor-pointer',
+            href: agreements ? '/' : undefined,
```
Comment (arrenv) [inline src/components/common/ColonyHome/ColonyHome.tsx:296]: We can use the `COLONY_AGREEMENTS_ROUTE` from `src/routes/routeConstants.ts`.

File: src/components/common/ColonyHome/hooks.ts [hunk 9]
```diff
@@ -94,12 +103,53 @@ export const useGetHomeWidget = (team?: number): UseGetHomeWidgetReturnType => {
...
+    ?.slice(3, allTeams.length)
+    .filter((item) => item.reputationPercentage !== null)
+    .reduce((acc, item) => acc + Number(item.reputationPercentage), 0);
+
+  const otherTeams = {
+    id: '4',
+    label: formatText({ id: 'label.allOther' }) ?? '',
+    value: otherTeamsReputation ?? 0,
+    color: '#F2F4F7',
+    stroke: '#F2F4F7',
```
Comment (arrenv) [inline src/components/common/ColonyHome/hooks.ts:123]: We should be using the Tailwind defined colours here.

File: src/components/v5/common/TeamReputationSummary/utils.ts [hunk 10]
```diff
@@ -39,6 +39,45 @@ export const setTeamColor = (color?: DomainColor) => {
...
+    case DomainColor.Periwinkle:
+      return '#4E5BA6';
+    case DomainColor.PurpleGrey:
+      return '#7A5AF8';
+    case DomainColor.Red:
+      return '#E31B54';
+    default:
+      return '#7A5AF8';
+  }
+};
```
Comment (arrenv) [inline src/components/v5/common/TeamReputationSummary/utils.ts:79]: Why are we not able to use the defined colours as opposed to defining again here with hexes.

File: src/components/v5/common/DonutChart/hooks.tsx [hunk 11]
```diff
@@ -0,0 +1,299 @@
...
+    () =>
+      data.length > 0 && summedChartValues > 1
+        ? data
+        : [
+            {
+              id: '1',
+              label: '',
+              value: 100,
+              color: '#F9FBFC',
+              stroke: '#F9FBFC',
```
Comment (arrenv) [inline src/components/v5/common/DonutChart/hooks.tsx:40]: Can we avoid using hexes and instead use the defined colour variables.

File: src/images/icons/message.svg
Comment (arrenv) [inline src/images/icons/message.svg:1]: Can we use phospher icons library instead of adding a new SVG with a hardcoded colour.

**Review by arrenv — CHANGES_REQUESTED** [review]
@krawczykkatarzyna Thank you for the updates, great work. A few comments.


1. The design shows the teams aligned vertically in the design.
image

This is what currently happens:

image


2. There is a padding at the bottom of a team, if there is no `All other`.

image

**Review by ArmandoGraterol — COMMENTED** [review]
I haven't spotted anything codewise but I'll take a closer look once the changes of the other reviews are in 👍

**Comment by krawczykkatarzyna** [comment]
@arrenv fixed

**Review by melyndav — APPROVED** [review]
@krawczykkatarzyna great work on this. All looks good from a UI standpoint!

**Review by arrenv — APPROVED** [review]
Everything picked up has been resolved, great work @krawczykkatarzyna thank you!

**Review by CzarekDryl — APPROVED** [review]

**Pull request merged**